app/controllers/users_controller.rb

```diff
@@ -36,7 +38,12 @@ def create
       render :new and return
     end
 
-    @user.role_id = Permissible::READONLY if params[:secret] != ENV["ACCOUNT_SECRET"]
+    if params[:secret] != ENV["ACCOUNT_SECRET"]
+      @user.role_id = Permissible::READONLY
+    elsif ENV.fetch("UPGRADES_LOCKED", nil).present?
+      flash.now[:error] = "We're sorry, full accounts are currently unavailable."
+      @user.role_id = Permissible::READONLY
+    end
```

Comment:
I was going to say I think these are in the wrong order and they might still be but only because of the flash, the first one is also setting reader